mifospay/src/main/java/org/mifospay/common/ui/MakeTransferScreen.kt

```diff
@@ -0,0 +1,411 @@
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import kotlinx.coroutines.launch
+import org.mifospay.R
+import org.mifospay.common.Constants
+import org.mifospay.common.presenter.MakeTransferState
+import org.mifospay.common.presenter.MakeTransferViewModel
+import org.mifospay.common.presenter.ShowTransactionStatus
+import org.mifospay.core.designsystem.component.MifosOverlayLoadingWheel
+
+@Composable
+fun MakeTransferScreenRoute(
+    viewModel: MakeTransferViewModel = hiltViewModel(),
+) {
+    val makeTransferState by viewModel.makeTransferState.collectAsStateWithLifecycle()
+    val showTransactionStatus by viewModel.showTransactionStatus.collectAsStateWithLifecycle()
+
+    MakeTransferScreen(
+        uiState = makeTransferState,
+        showTransactionStatus = showTransactionStatus,
+        makeTransfer = { toClientId, transferAmount ->
+            viewModel.makeTransfer(
+                toClientId,
+                transferAmount
+            )
+        }
+    )
+}
+
+@Composable
+fun MakeTransferScreen(
+    uiState: MakeTransferState,
+    showTransactionStatus: ShowTransactionStatus,
+    makeTransfer: (Long, Double) -> Unit,
+) {
+    val context = LocalContext.current
+    when (uiState) {
+        MakeTransferState.Loading -> {
+            MifosOverlayLoadingWheel(contentDesc = stringResource(R.string.loading))
+        }
+
+        is MakeTransferState.Error -> {
+            val message = uiState.message
+            Toast.makeText(context, message, Toast.LENGTH_SHORT).show()
+        }
+
+        is MakeTransferState.Success -> {
+
+            val showBottomSheet = uiState.showBottomSheet
+            val toClientId = uiState.toClientId
+            val resultName = uiState.resultName
+            val externalId = uiState.externalId
+            val transferAmount = uiState.transferAmount
+            val showTransactionStatus = showTransactionStatus
+
+            MakeTransferBottomSheetContent(
+                showBottomSheet,
+                toClientId,
+                resultName,
+                externalId,
+                transferAmount,
+                showTransactionStatus,
+                makeTransfer = makeTransfer
+            )
```

Comment:
directly pass these values to the composable. 
also always pass by naming the parameter, link `showBottomSheet = uiState.showBottomSheet` for functions for more good visibility of code.